Earthfile

```diff
@@ -0,0 +1,400 @@
+# See https://docs.earthly.dev/docs/earthfile/features
+VERSION --try --raw-output --run-with-aws 0.8
+
+PROJECT crossplane/crossplane
+
+ARG --global GO_VERSION=1.22.3
+
+# reviewable checks that a branch is ready for review. Run it before opening a
+# pull request. It will catch a lot of the things our CI workflow will catch.
+reviewable:
+  WAIT
+    BUILD +generate
+  END
+  BUILD +lint
+  BUILD +test
+
+# test runs unit tests.
+test:
+  BUILD +go-test
+
+# lint runs linters.
+lint:
+  BUILD +go-lint
+  BUILD +helm-lint
+
+# build builds Crossplane for your native OS and architecture.
+build:
+  BUILD +image
+  BUILD +helm-build
+
+# multiplatform-build builds Crossplane for all supported OS and architectures.
+multiplatform-build:
+  BUILD +go-multiplatform-build
+  BUILD +multiplatform-image
+  BUILD +helm-build
+
+# generate runs code generation. To keep builds fast, it doesn't run as part of
+# the build target. It's important to run it explicitly when code needs to be
+# generated, for example when you update an API type.
+generate:
+  BUILD +go-modules-tidy
+  BUILD +go-generate
+  BUILD +helm-generate
+
+# e2e runs end-to-end tests. See test/e2e/README.md for details. 
+e2e:
+  ARG --required FLAGS
+  # Docker installs faster on Alpine, and we only need Go for go tool test2json.
+  FROM golang:${GO_VERSION}-alpine3.20
+  RUN apk add --no-cache docker jq
+  COPY +helm-setup/helm /usr/local/bin/helm
+  COPY +kind-setup/kind /usr/local/bin/kind
+  COPY +gotestsum-setup/gotestsum /usr/local/bin/gotestsum
+  COPY +go-build-e2e/e2e .
+  COPY --dir cluster test .
+  WITH DOCKER --load crossplane-e2e/crossplane:latest=+image
+    TRY
+      # TODO(negz:) Set GITHUB_ACTIONS=true and use RUN --raw-output when
+      # https://github.com/earthly/earthly/issues/4143 is fixed.
+      RUN gotestsum --no-color=false --format testname --junitfile e2e-tests.xml --raw-command go tool test2json -t -p E2E ./e2e -test.v ${FLAGS}
+    FINALLY
+      SAVE ARTIFACT --if-exists e2e-tests.xml AS LOCAL _output/tests/e2e-tests.xml
+    END
+  END
+
+# hack builds Crossplane, and deploys it to a kind cluster. It runs in your
+# local environment, not a container. The kind cluster will keep running until
+# you run the unhack target. Run hack again to rebuild Crossplane and restart
+# the kind cluster with the new build.
+hack:
```

Comment:
I think the problem was that `TARGETOS` binaries won't work for `LOCALLY` targets, since `TARGETOS` will always be Linux. I've pushed a commit that I hope will fix this. Verified it still works when run from Linux.